apps/condo/domains/ticket/schema/TicketFile.test.js

```diff
@@ -0,0 +1,164 @@
+        const client = await makeClientWithProperty()
+        const [ticketFile, attrs] = await createTestTicketFile(client, client.organization)  
+        expect(ticketFile.id).toMatch(UUID_RE)
+        expect(ticketFile.dv).toEqual(1)
+        expect(ticketFile.sender).toEqual(attrs.sender)
+        expect(ticketFile.v).toEqual(1)
+        expect(ticketFile.newId).toEqual(null)
+        expect(ticketFile.deletedAt).toEqual(null)
+        expect(ticketFile.createdBy).toEqual(expect.objectContaining({ id: client.user.id }))
+        expect(ticketFile.updatedBy).toEqual(expect.objectContaining({ id: client.user.id }))
+        expect(ticketFile.createdAt).toMatch(DATETIME_RE)
+        expect(ticketFile.updatedAt).toMatch(DATETIME_RE)
+        expect(ticketFile.organization).toEqual(expect.objectContaining({ id: client.organization.id }))
+    })
+
+
+    test('anonymous: create TicketFile', async () => {
+        const client = await makeClient()
+        const clientWithOrganization = await makeClientWithProperty()
+        try {
+            await createTestTicketFile(client, clientWithOrganization.organization) 
+        } catch (e) {
+            expect(e.errors[0]).toMatchObject({
+                'message': 'You do not have access to this resource',
+                'name': 'AccessDeniedError',
+                'path': ['obj'],
+            })
+            expect(e.data).toEqual({ 'obj': null })
+        }
+    })
+
+    test('user: read TicketFile', async () => {
+        const client = await makeClientWithTicket()
+        const [ticketFile, attrs] = await createTestTicketFile(client, client.organization, client.ticket)  
+        const objs = await TicketFile.getAll(client, {}, { sortBy: ['updatedAt_DESC'] })
+        expect(objs).toHaveLength(1)
+        expect(objs[0].id).toMatch(ticketFile.id)
+        expect(objs[0].dv).toEqual(1)
+        expect(objs[0].sender).toEqual(attrs.sender)
+        expect(objs[0].v).toEqual(1)
+        expect(objs[0].newId).toEqual(null)
+        expect(objs[0].deletedAt).toEqual(null)
+        expect(objs[0].createdBy).toEqual(expect.objectContaining({ id: client.user.id }))
+        expect(objs[0].updatedBy).toEqual(expect.objectContaining({ id: client.user.id }))
+        expect(objs[0].createdAt).toMatch(ticketFile.createdAt)
+        expect(objs[0].updatedAt).toMatch(ticketFile.updatedAt)
+    })
+
+    test('anonymous: read TicketFile', async () => {
+        const client = await makeClient()
+        try {
+            await TicketFile.getAll(client)
+        } catch (e) {
+            expect(e.errors[0]).toMatchObject({
+                'message': 'You do not have access to this resource',
+                'name': 'AccessDeniedError',
+                'path': ['objs'],
+            })
+            expect(e.data).toEqual({ 'objs': null })
+        }
+    })
+
+    test('user: read TicketFile from another organization', async () => {
```

Comment:
I'ts easier to read and understand test cases, when they are grouped to `describe` blocks.

For example:

```
describe('user: read TicketFile', () => {
    it('cannot read tickets from another organization')
    it('can read tickets from organization, it employed in')
})
```

Reply:
Done

Reply:
I meant it to be different.

When we strip a code at `759d1e65` and read it, like an outline of a book, we have:
TicketFile
  - Create
    - it User create temporary TicketFile [no ticket relation]
    - it User create TicketFile
    - it Anonymous have no access to create TicketFile

It's unclear, it can or cannot do something and why.

Compare with the following outline:

TicketFile
  - user: read TicketFile
    - it cannot read ticket files from another organization
    - it can read ticket files from organization, it employed in
    - it cannot read ticket files, when it's blocked
    - …

Is it better to read, is't?